test/scenarios/el92-src@rpm-install.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+
+# shellcheck source=test/bin/get_rel_version_repo.sh
+source "${SCRIPTDIR}/get_rel_version_repo.sh"
+
+# Sourced from scenario.sh and uses functions defined there.
+
+# The RPM-based system won't run greenboot automatically on startup,
+# so tell the framework not to wait for that.
+export SKIP_GREENBOOT=true
+
+# NOTE: Unlike most suites, these tests rely on being run IN ORDER to
+# ensure the host is in a good state at the start of each test. We
+# could have separated them and run them as separate scenarios, but
+# did not want to spend the resources on a new VM.
+export TEST_RANDOMIZATION=none
+
+scenario_create_vms() {
+    prepare_kickstart host1 kickstart-liveimg.ks.template ""
+    launch_vm host1 "rhel-9.2"
+
+    # Open the firewall ports. Other scenarios get this behavior by
+    # embedding settings in the blueprint, but there is no blueprint
+    # for this scenario. We need do this step before running the RF
+    # suite so that suite can assume it can reach all of the same
+    # ports as for any other test.
+    configure_vm_firewall host1
+
+    # Register the host with subscription manager so we can install
+    # dependencies.
+    subscription_manager_register host1
+}
+
+scenario_remove_vms() {
+    remove_vm host1
+}
+
+scenario_run_tests() {
+    set -x
+    local -r source_reponame=$(basename "${LOCAL_REPO}")
+
+    # Determine the version of the RPM in the local repo so we can use it
+    # in the blueprint templates.
+    if [ ! -d "${LOCAL_REPO}" ]; then
+        error "Run ${SCRIPTDIR}/create_local_repo.sh before building images."
+        return 1
+    fi
+    local -r release_info_rpm=$(find "${LOCAL_REPO}" -name 'microshift-release-info-*.rpm' | sort | tail -n 1)
+    if [ -z "${release_info_rpm}" ]; then
+        error "Failed to find microshift-release-info RPM in ${LOCAL_REPO}"
+        return 1
+    fi
+    local -r release_info_rpm_base=$(find "${BASE_REPO}" -name 'microshift-release-info-*.rpm' | sort | tail -n 1)
+    if [ -z "${release_info_rpm_base}" ]; then
+        error "Failed to find microshift-release-info RPM in ${BASE_REPO}"
+        return 1
+    fi
+    local -r SOURCE_VERSION=$(rpm -q --queryformat '%{version}' "${release_info_rpm}")
+    local -r CURRENT_VERSION=$(echo "${SOURCE_VERSION}" | cut -f1-2 -d.)
+    local -r MINOR_VERSION=$(echo "${CURRENT_VERSION}" | cut -f2 -d.)
+    local -r PREVIOUS_MINOR_VERSION=$(( "${MINOR_VERSION}" - 1 ))
```

Comment:
This code already exists in `build_images.sh` and partially in `get_latest_rpm_version.sh`.
It's an implementation detail. 
Should we abstract it from the test by putting it in a separate script and calling it in those scripts and also in the test?

Reply:
I cleaned this up.